src/main/java/picard/util/ThreadPoolExecutorWithExceptions.java

```diff
@@ -40,6 +42,7 @@ protected void afterExecute(final Runnable r, Throwable t) {
             }
         }
         if (t != null) {
+            exception = t;
             throw new PicardException(t.getMessage(), t);
```

Comment:
Is there no way to get the exception out of the thrown exception here?

Reply:
The executor doesn't expose exceptions thrown by its runnables. This is the only hook that allows you to grab them. Though I don't see why we couldn't just shutdown the executor on error.

Reply:
actually now that I think about it this is a signal to a completion executor so we need to know if we exiting normally or with an exception.. I'll play around with it a bit.